File: packages/maplibre_flutter_android/android/src/main/kotlin/dev/maplibreflutter/maplibre_flutter_android/MapLibreViewFactory.kt
package dev.maplibreflutter.maplibre_flutter_android

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/**
 * Builds a [MapLibrePlatformView] per `AndroidView`. The creation args are the
 * `creationParams` map the Dart controller serialises from `MapOptions`
 * (styleUri + camera), decoded via [StandardMessageCodec].
 */
class MapLibreViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        @Suppress("UNCHECKED_CAST")
        val params = args as? Map<String, Any?> ?: emptyMap()
        return MapLibrePlatformView(context, params)
    }
}

File: packages/maplibre_flutter_ios_sdk/ios/maplibre_flutter_ios_sdk/Sources/maplibre_flutter_ios_sdk/MapLibreViewFactory.swift
import Flutter
import UIKit

/// Builds a `MapLibrePlatformView` per `UiKitView`. The creation args are the
/// `creationParams` map the Dart controller serialises from `MapOptions`
/// (`mapId`, `styleUri`, camera), decoded via `FlutterStandardMessageCodec`.
class MapLibreViewFactory: NSObject, FlutterPlatformViewFactory {
  private let messenger: FlutterBinaryMessenger

  init(messenger: FlutterBinaryMessenger) {
    self.messenger = messenger
    super.init()
  }

  func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
    FlutterStandardMessageCodec.sharedInstance()
  }

  func create(
    withFrame frame: CGRect,
    viewIdentifier viewId: Int64,
    arguments args: Any?
  ) -> FlutterPlatformView {
    MapLibrePlatformView(frame: frame, args: args)
  }
}
